src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Patch_V1Compat.txt

```diff
@@ -39,5 +52,242 @@ Errors: Error 13-143: The value passed to the 'Patch' function cannot be changed
+Errors: Error 6-9: Name isn't valid. 'Foo' isn't recognized.|Error 0-52: The function 'Patch' has some invalid arguments.|Error 11-20: The specified column 'Field1' does not exist.|Error 22-51: The specified column 'DisplayNameField2' does not exist.
+
+>> Patch(Foo, Bar, {DisplayNameField2:"jupiter"})
+Errors: Error 6-9: Name isn't valid. 'Foo' isn't recognized.|Error 11-14: Name isn't valid. 'Bar' isn't recognized.|Error 0-46: The function 'Patch' has some invalid arguments.|Error 16-45: The specified column 'DisplayNameField2' does not exist.
+
+>> Set(t_an_bs, [{a:1, b:"a"}, {a:2, b:Blank()}]); Patch(t_an_bs, Last(t_an_bs), {a:3, b:"c"}); t_an_bs
+Table({a:1,b:"a"},{a:3,b:"c"})
+
+>> Set(t_an_bs, [{a:1,b:If(false,"")}, {a:Blank(),b:"1"}]); Patch(t_an_bs, Last(t_an_bs), {b:"2"}); t_an_bs
+Table({a:1,b:Blank()},{a:Blank(),b:"2"})
+
+>> Set(t_an_bs, Table({a:1,b:If(false,"")},{a:Blank(),b:"2"})); Patch(t_an_bs, Last(t_an_bs), {a:2}); t_an_bs
+Table({a:1,b:Blank()},{a:2,b:"2"})
+
+
+// Base record not found. If the record in its entirety is not matched then it should produce an error. 
+>> Collect(t1,{Field1:3,Field2:"phobos",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false});
+   Collect(t1,{Field1:2,Field2:"deimos",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false});
+   Patch(t1,{Field2:"deimos"},{Field3:DateTime(2030,2,1,0,0,0,0)})
+Error({Kind:ErrorKind.NotFound})
+
+>> Collect(t1,{Field1:3,Field2:"phobos",Field3:DateTime(2024,2,1,0,0,0,0),Field4:false});
+   Collect(t1,{Field1:2,Field2:"deimos",Field3:DateTime(2025,2,1,0,0,0,0),Field4:false});
+   Patch(t1,{Field3:DateTime(2025,2,1,0,0,0,0)},{Field2:"pandora", Field1:55})
+Error({Kind:ErrorKind.NotFound})
+
+>> Collect(t1, r2);
+   Patch(t1, {Field4:true}, {Field2:"phobos"});
+   First(t1).DisplayNameField2
+Error({Kind:ErrorKind.NotFound})
+
+>> Patch(t1, {Field1:1}, {DisplayNameField2:"mars"});
+   First(t1).Field2
+Error({Kind:ErrorKind.NotFound})
+
+>> Patch(t1, {Field1:55}, {DisplayNameField2:"mars"})
+Error({Kind:ErrorKind.NotFound})
+
+>> Patch(t1, {DisplayNameField1:55}, {DisplayNameField2:"mars"})
+Error({Kind:ErrorKind.NotFound})
+
+>> Patch(t1, {DisplayNameField1:1/0}, {DisplayNameField2:"mars"})
+Error({Kind:ErrorKind.NotFound})
+
+>> Patch(t1, {DisplayNameField2:"earth"}, {Field2:"mars"});
+   t1
+Error({Kind:ErrorKind.NotFound})
+
+>> Patch(Blank(),Blank(),Blank())
+Blank()
+
+>>Patch(If(false,Table()),{},{})
+Errors: Error 6-23: The value passed to the 'Patch' function cannot be changed.
+
+>> ForAll(Sequence(5), Collect(t_empty, {Value:ThisRecord.Value})); ForAll(Sequence(CountRows(t_empty)), Collect(t_empty2, Index(t_empty, ThisRecord.Value))); Patch(t_empty, First(t_empty), {Value:99}); Patch(t_empty2, Last(t_empty2), {Value:88}); Concat(t_empty, Value) & "," & Concat(t_empty2, Value)
+"992345,123488"
+
+
+
+// **********************************************************
+// Patch(Record, Updates1, Updates2,…)
+// **********************************************************
```

Comment:
Consider adding more tests with literal records, and with some scenarios where this is interesting.

Some scenario examples:

- Adding an index to a table:

    With(
      { tbl:Table({a:1,b:2}, {a:11,b:22}, {a:111,b:222}) }
      ForAll(
        Sequence(CountRows(tbl)),
        Patch(
          { Index: ThisRecord.Value },
          Index(tbl, ThisRecord.Value)
        )
      )
    )

- Flattening record fields

    With(
      { rec: { Name: "John Doe", Address: { Country: "USA", State: "WA", City: "Seattle", Street: "85 Pike Street" } } },
      Patch(
        DropColumns(rec, Address),
        rec.Address
      )
    )